### PR TITLE
Advanced Mode: collect negative feedback internally and get ratings from positive users on app store

### PR DESCRIPTION
**Pull request by harshabonthu**
Hi,

We've been using this advanced mode to improve our app ratings and noticed major positive rating improvements on the app store with below flow.

1) Instead of directly asking our users to Rate our app, we check if the user likes using our app. so the first popup would be
`Do you like using {{APP NAME}}`

2) If the user says 'Yes' then we ask the user if they would like take a moment and rate our app (Current default dialog shown with default options like Yes, Maybe later, No).
If the user says 'NO' to Question 1, We ask the user another question - `Would you mind providing us feedback`

3) If the user says yes, then we redirect them to a url or a email id to collect feedback.

I didn't yet update docs to reflect advanced mode settings. Just wanted to get the conversation started and update based on feedback.

To try these changes, below is a sample config I use

```
    AppRate.preferences.promptAgainForEachNewVersion = true;
    AppRate.preferences.storeAppURL.ios = '93xxxxx';
    AppRate.preferences.storeAppURL.android = 'market://details?id=com.xxxxxxxx';
    AppRate.preferences['isAdvancedMode'] = true;
    AppRate.preferences.usesUntilPrompt = 5;
    AppRate.preferences['appFeedbackURL'] = 'mailto:support@example.com'; // Can be a url too
    AppRate.promptForRating(true);
```

Thanks.

### CODE REVIEW

**Comment by westonganger** [comment]
Sure we can add this however this PR needs to include documentation for the new feature in the README

**Comment by fobaz** [comment]
Is it possible to get this added?

**Comment by remisture** [comment]
Love to have this feature!

File: www/AppRate.js [hunk 1]
```diff
@@ -168,6 +210,7 @@ AppRate = (function() {
   AppRate.preferences = {
     useLanguage: null,
     displayAppName: '',
+    isAdvancedMode: true,
```
Comment (westonganger) [inline www/AppRate.js]: I think that this "advanced mode" should be the default, and we should instead have `simpleMode: false` here

File: www/AppRate.js [hunk 2]
```diff
@@ -184,6 +227,7 @@ AppRate = (function() {
       windows8: null,
       windows: null
     },
+    appFeedbackURL: '/#',
```
Comment (westonganger) [inline www/AppRate.js]: If no feedback url is supplied, it should not ask the user for feedback when "No" is selected. I cant tell if this is the case already. Or maybe we could have an option `feedbackEnabled` or whatever

**Comment by harshabonthu** [comment]
Sorry, its been a while. Its just the docs thats missing right now,  I can work on this today evening.

**Comment by sam2x** [comment]
Great idea. would be great if we could handle the "negative feedback" via our own app. For example, passing a function callback to appFeedbackURL, and using dialog inside the app. My idea is to show a popup that has a list of checkbox for negative reasons (ex: performance, design, ergonomy, etc.) , a text input for optional description, and a submit button).

**Comment by westonganger** [comment]
@sam2x I agree a custom negative callback should definitely be added too.

**Comment by harshabonthu** [comment]
updated PR based on feedback.

- Added option for simpleMode to fallback to previous flow
- Removed AppfeedbackURL and instead added option for callback method `handleAppFeedback`
- Fallbacks to simplemode if the callback `handleAppFeedback` is not defined.

**Comment by westonganger** [comment]
Sorry to be picky but does `handleAppFeedback` make sense? I think `handleNegativeFeedback` or something along those lines is more appropriate.

**Comment by harshabonthu** [comment]
@westonganger  Agree, updated to handleNegativeFeedback

**Comment by westonganger** [comment]
Merged! Great work, thanks for your contribution.

**Pull request merged**